src/common/Gateway.sol

```diff
@@ -249,32 +250,33 @@ contract Gateway is Auth, IGateway, IRecoverable {
     function send(uint16 chainId, bytes calldata message) external pauseable auth {
         require(message.length > 0, "Gateway/empty-message");
 
+        PoolId poolId = processor.messagePoolId(message);
         if (isBatching) {
```

Comment:
Could we not remove `isBatching` and just use `batchLocators.length > 0`?

Reply:
Totally! 🚀 Good catch!

Reply:
Finally we cannot do that 😢 , given we need to signal if the batching has started in `startBatching()` when still no messages have been sent.

Reply:
Right :(